test/e2e/e2e_test.go

```diff
@@ -296,8 +298,18 @@ func (o *options) DefaultClusterOptions(t *testing.T) core.CreateOptions {
 		// align with default for e2e.aws-region flag
 		createOption.AWSPlatform.Zones = []string{"us-east-1a"}
 	} else {
-		createOption.AWSPlatform.Zones = strings.Split(o.configurableClusterOptions.Zone.String(), ",")
-		createOption.AzurePlatform.AvailabilityZones = strings.Split(o.configurableClusterOptions.Zone.String(), ",")
+		// For AWS, select a single zone for InfrastructureAvailabilityPolicy: SingleReplica guest cluster.
+		// This option is currently not configurable through flags and not set manually
+		// in any test, so we know InfrastructureAvailabilityPolicy is SingleReplica.
+		// If any test changes this in the future, we need to add logic here to make the
+		// guest cluster multi-zone in that case.
+		zones := strings.Split(o.configurableClusterOptions.Zone.String(), ",")
+		awsGuestZone := zones[nextAWSZoneIndex]
+		nextAWSZoneIndex = (nextAWSZoneIndex + 1) % len(zones)
+		createOption.AWSPlatform.Zones = []string{awsGuestZone}
```

Comment:
@sjenning I don't think we can make this assumption for all hosted clusters. For example the upgrade test is done with an HA control plane: https://github.com/openshift/hypershift/blob/main/test/e2e/control_plane_upgrade_test.go#L28-L30
Maybe the control plane availability setting should be a parameter to this function and only do the single zone thing if you want a SingleReplica cluster.